student-os/scripts/materials_convert.py

```
@@ -461,6 +468,165 @@ def get_mineru_client(ctx: ConversionContext) -> Any:
+        with urllib.request.urlopen(request, timeout=timeout) as response:
+            response.read()
+    except urllib.error.HTTPError as exc:
+        detail = exc.read().decode("utf-8", errors="replace").strip()
+        raise RuntimeError(f"MinerU v1 Agent upload HTTP {exc.code}: {detail}") from exc
+    except urllib.error.URLError as exc:
+        raise RuntimeError(f"MinerU v1 Agent upload failed: {exc.reason}") from exc
+
+
+def http_text(url: str, *, timeout: int = 300) -> str:
+    class NoRedirectHandler(urllib.request.HTTPRedirectHandler):
+        def redirect_request(self, req, fp, code, msg, headers, newurl):  # type: ignore[no-untyped-def]
```

Ruff (0.16.1)

[warning] Missing return type annotation for private function `redirect_request`

Add return type annotation: `None`

(ANN202)